ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplayFeatureLayers/DisplayFeatureLayers.qml

```diff
@@ -0,0 +1,215 @@
+
+    // Display a feature layer with a Geodatabase
+    FeatureLayer {
+        id: gdbFeatureLayer
+        featureTable: geodatabase.geodatabaseFeatureTablesByTableName["Trailheads"] ?? null
+        Geodatabase {
+            id: geodatabase
+            path: dataPath + "geodatabase/LA_Trails.geodatabase"
+        }
+    }
+
+    function setGeodatabaseLayer() {
+        map.operationalLayers.append(gdbFeatureLayer);
+
+        // Set the viewpoint to the layer
+        const viewpointCenterPoint = ArcGISRuntimeEnvironment.createObject("Point", {x: -13214155, y: 4040194, spatialReference: Factory.SpatialReference.createWebMercator()});
+        mapView.setViewpointCenterAndScale(viewpointCenterPoint, 35e4);
+    }
+
+    // Display a GeoPackage feature layer
+    FeatureLayer {
+        id: gpkgFeatureLayer
+    }
+
+    GeoPackage {
+        id: gpkg
+        path: dataPath + "gpkg/AuroraCO.gpkg"
+
+        // Wait for the GeoPackage to load successfully
+        onLoadStatusChanged: {
+            if (loadStatus !== Enums.LoadStatusLoaded || // the GeoPackage hasn't yet loaded
+                    featureLayerComboBox.currentText !== "GeoPackage" || // the user selected a different option while loading and this process should be aborted
+                    gpkg.geoPackageFeatureTables.length === 0) // the GeoPackage has no feature tables to use
+                return;
+
+            setGeoPackageLayer();
+        }
+
+        onErrorChanged: {
+            console.log(error.message, error.additionalMessage);
+        }
+    }
+
+    function setGeoPackageLayer() {
+        if (gpkg.loadStatus !== Enums.LoadStatusLoaded) {
+            // The GeoPackage must be explicitly loaded before its feature tables can be accessed
+            gpkg.load();
+            return;
+        }
+
+        // A GeoPackageFeatureTable can only be accessed by one feature layer at a time
+        if (!gpkgFeatureLayer.featureTable) {
+            // Get the first feature table
+            const gpkgFeatureTable = gpkg.geoPackageFeatureTables[0];
+            // Create a FeatureLayer with the table
+            gpkgFeatureLayer.featureTable = gpkgFeatureTable;
+        }
+
+        // Append the layer to the map
+        map.operationalLayers.append(gpkgFeatureLayer);
+
+        // Set the viewpoint to the layer
+        const viewpointCenterPoint = ArcGISRuntimeEnvironment.createObject("Point", {x: -104.8319, y: 39.7294, spatialReference: Factory.SpatialReference.createWgs84()});
+        mapView.setViewpointCenterAndScale(viewpointCenterPoint, 2e5);
+
+        // Note you must call close() on GeoPackage to allow other processes to access it
+    }
+
+    // Display a feature layer with a portal item
+    FeatureLayer {
+        id: portalItemFeatureLayer
+        item: PortalItem {
+            itemId: "1759fd3e8a324358a0c58d9a687a8578"
+        }
+    }
+
+    function setPortalItemLayer() {
+        map.operationalLayers.append(portalItemFeatureLayer);
```

Comment:
what happens when you cycle through the combo box and end up selecting all of the items more than once? The C++ version of this sample creates new objects, this one uses declared objects that it won't be able to add to the map twice

Reply:
`operationalLayers.clear()` is called on line 74 before setting a new layer, and I confirmed they're successfully removed and added back when their respective methods are called subsequent times.